src/main/scala/codacy/stylelint/DocGenerator.scala

```diff
@@ -63,7 +63,10 @@ object DocGenerator {
     val patternsDescription: Set[Pattern.Description] = patterns.map { patternid =>
       val patternDescription =
         ParseMarkupRule.parseForDescriptions(File(rulesdir + "/" + patternid + "/README.md"))
-      addNewDescription(patternid, patternDescription)
+
+      val urlRegex = """\[(.+)]\((.+)\)""".r
```

Comment:
This regex seems a bit strange to me, what are you trying to match and catch? I don't understand the "\[" and the ] that does not seem to be escaped? Also consider adding a comment of what we want to catch with example.

Also, I think regexes with capture groups become clear with pattern matching like:

```scala
scala> val MyRegex = """(.+)/(.+)""".r
MyRegex: scala.util.matching.Regex = (.+)/(.+)

scala> val input = "hey/bye"
input: String = hey/bye

scala> input match {
     |   case MyRegex(batata, melancia) => s"Extracted '$batata' and '$melancia'"
     |   case _ => s"Didn't extract anything"
     | }
res0: String = Extracted 'hey' and 'bye'
```

Never used with replaceAllIn, but what is the second argument there? Can you pattern match there? With like `case urlRegex(x) => x`, then we would even have a name for x, and I would semantically know what is group(1) :D

Reply:
As far as I know, I can't do it with the replaceAllIn